crates/shim-sgx/src/handler/mod.rs

```diff
@@ -109,6 +104,108 @@ impl<'a> Handler<'a> {
                 .unwrap_or_else(|_| self.attacked());
         }
     }
+
+    fn mprotect_unlocked(
+        &mut self,
+        heap: &mut Heap,
+        addr_in: NonNull<c_void>,
+        length_in: c_size_t,
+        prot: c_int,
+    ) -> sallyport::Result<()> {
+        let addr = addr_in.as_ptr() as usize;
+        let pages = ((length_in + Page::SIZE - 1) & !(Page::SIZE - 1)) / Page::SIZE;
```

Comment:
I find this calculation extremely cryptic, can we get a comment? Why are we taking the bitwise-and of a division of a bitwise-negation of one less than the page size? Some parentheses to make the grouping clearer would also be nice, or maybe extract it into a function and document it there.

Reply:
It should've been probably `(length_in + (Page::SIZE - 1)) / Page::SIZE` but it would be wrong to change the code in this commit because the point is to moved tested code as it is to a new place, not to improve it.

Reply:
IMHO, patches should not only be as localized as possible in terms SLOC but also in terms of logical changes made.

Reply:
Can you add a FIXME comment here on improving this in the future?

Reply:
`FIXME` would be a wrong choice because it would imply there was, which does not exist, so I'll add `TODO` instead.

Reply:
I think this should be sufficient here because there is nothing functionally broken in the code:

```rust
        // TODO: Simplify:
        let pages = ((length_in + Page::SIZE - 1) & !(Page::SIZE - 1)) / Page::SIZE;
```

It's also too low priority, given that the compiled code will be the same, that it does not make sense to make an issue for it. It will be fixed eventually when there is some need to change this code otherwise.